llama-index-integrations/llms/llama-index-llms-azure-inference/llama_index/llms/azure_inference/base.py

```diff
@@ -279,11 +282,22 @@ def class_name(cls) -> str:
     @property
     def metadata(self) -> LLMMetadata:
         if not self._model_name:
-            model_info = self._client.get_model_info()
-            if model_info:
-                self._model_name = model_info.get("model_name", None)
-                self._model_type = model_info.get("model_type", None)
-                self._model_provider = model_info.get("model_provider_name", None)
+            try:
+                # Get model info from the endpoint. This method may not be supported by all
+                # endpoints.
+                model_info = self._client.get_model_info()
+                if model_info:
+                    self._model_name = model_info.get("model_name", None)
+                    self._model_type = model_info.get("model_type", None)
+                    self._model_provider = model_info.get("model_provider_name", None)
+            except Exception:
```

Comment:
ooc is there a more specific Exception we can catch?

Reply:
Good catch. I replaced this by `HttpResponseError`.